### ISSUE TITLE
Build error, v.0.3.2

### ISSUE THREAD

**Issue by hombit** [chalharu/chfft#34]
I'm trying to build a crate depended on `chfft` and have this error:
```
error[E0308]: mismatched types
  --> /Users/hombit/.cargo/registry/src/github.com-1ecc6299db9ec823/chfft-0.3.2/src/precompute_utils.rs:18:9
   |
16 | pub fn calc_omega_item<T: Float + FloatConst>(len: usize, position: usize) -> Complex<T> {
   |                        - this type parameter
17 |     Complex::from_polar(
18 |         &one(),
   |         ^^^^^^
   |         |
   |         expected type parameter `T`, found reference
   |         help: consider removing the borrow: `one()`
   |
   = note: expected type parameter `T`
                   found reference `&_`
   = help: type parameters must be constrained to match other types
   = note: for more information, visit https://doc.rust-lang.org/book/ch10-02-traits.html#traits-as-parameters

error[E0308]: mismatched types
  --> /Users/hombit/.cargo/registry/src/github.com-1ecc6299db9ec823/chfft-0.3.2/src/precompute_utils.rs:19:9
   |
16 | pub fn calc_omega_item<T: Float + FloatConst>(len: usize, position: usize) -> Complex<T> {
   |                        - this type parameter
...
19 |         &(cast::<_, T>(-2.0).unwrap() * T::PI() / cast(len).unwrap() * cast(position).unwrap()),
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |         |
   |         expected type parameter `T`, found `&T`
   |         help: consider removing the borrow: `(cast::<_, T>(-2.0).unwrap() * T::PI() / cast(len).unwrap() * cast(position).unwrap())`
   |
   = note: expected type parameter `T`
                   found reference `&T`
   = help: type parameters must be constrained to match other types
   = note: for more information, visit https://doc.rust-lang.org/book/ch10-02-traits.html#traits-as-parameters

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
error: could not compile `chfft`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by hombit** [comment]
Previously compilation worked, can it be a problem with update `num-complex` to version 0.3?

**Comment by chalharu** [comment]
Thank you. Fixed in 0.3.4.
